packages/contracts-periphery/contracts/universal/op-nft/CitizenshipChecker.sol

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { SocialContract } from "./SocialContract.sol";
+
+interface ICitizenshipChecker {
+    function isCitizen(address _who, bytes memory _proof) external view returns (bool);
+}
+
+contract CitizenshipChecker is ICitizenshipChecker {
+    address public root;
+
+    SocialContract public sc;
+
+    constructor(address _root, address sca) {
+        root = _root;
+        sc = SocialContract(sca);
+    }
+
+    function isCitizen(address _who, bytes memory _proof) public view returns (bool) {
+        (address opco, uint256 index) = abi.decode(_proof, (address, uint256));
+        bytes memory numOpcoCitizenships = sc.attestations(root, opco, keccak256("op.opco"));
+        require(
+            index < abi.decode(numOpcoCitizenships, (uint256)),
+            "CitizenshipChecker::isCitizen: INVALID_INDEX"
```

Comment:
We generally don't include the function name in the `require` statement as part of our style, cc @smartcontracts